### PR TITLE
Token migration

### PR DESCRIPTION
**Pull request by mgloystein**
DO NOT MERGE! Please review only, we are not planning on pushing this back to Ory.

### CODE REVIEW

File: handler/oauth2/strategy_hmacsha.go [hunk 1]
```diff
@@ -39,7 +40,13 @@ func (h HMACSHAStrategy) ValidateAccessToken(_ context.Context, r fosite.Request
 	if !exp.IsZero() && exp.Before(time.Now()) {
 		return errors.Wrap(fosite.ErrTokenExpired, fmt.Sprintf("Access token expired at %s", exp))
 	}
-	return h.Enigma.Validate(token)
+
+	if err = h.Enigma.Validate(token); err != nil {
+		if split := strings.Split(token, "."); len(split) == 2 && split[0] == split[1] {
```
Comment (mooyah) [inline handler/oauth2/strategy_hmacsha.go]: this line is a bit mysterious. A comment could help.

File: introspection_request_handler.go [hunk 2]
```diff
@@ -121,7 +121,9 @@ func (f *Fosite) NewIntrospectionRequest(ctx context.Context, r *http.Request, s
 			return &IntrospectionResponse{Active: false}, errors.Wrap(ErrRequestUnauthorized, "HTTP Authorization header missing, malformed or credentials used are invalid")
 		}
 	}
-
+	// if r.PostForm.Get("migrated") == "true" {
```
Comment (mooyah) [inline introspection_request_handler.go]: planning on removing this commented code?

**Review by mooyah — COMMENTED** [review]
Overall seems OK, but I'm missing exactly how the migration occurs.

File: introspection_request_handler.go (hunk 2 above)
Comment (mgloystein) [reply introspection_request_handler.go]: Yes

File: migration_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
+package fosite_test
+
+import (
+	//"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/ory/fosite"
```
Comment (mooyah) [inline migration_handler_test.go]: is that "." intentional?

Comment (mgloystein) [reply migration_handler_test.go]: Yup, in Go that means to import that package and treat it as the current package.

**Comment by mgloystein** [comment]
Will not be merging to main branch

**Pull request closed without merging**